modules/augain/augain.c

```diff
@@ -0,0 +1,156 @@
+/**
+ * @file augain.c  Audio gain
+ *
+ * Copyright (C) 2024 Juha Heinanen
+ */
+
+#include <stdlib.h>
+#include <re.h>
+#include <rem.h>
+#include <baresip.h>
+
+
+/**
+ * Audio gain module
+ *
+ * This module can be used to increase volume of audio source,
+ * for example, microphone.
+ *
+ * Sample config:
+ *
+ \verbatim
+  augain            1.5
+ \endverbatim
+ */
+
+
+struct augain_enc {
+	struct aufilt_enc_st af;  /* base class */
+};
+
+
+static double gain = 1.0;
+
+
+static int conf_get_float(const struct conf *conf, const char *name,
+			  double *val)
+{
+	struct pl opt;
+	int err;
+
+	if (!conf || !name || !val)
+		return EINVAL;
+
+	err = conf_get(conf, name, &opt);
+	if (err)
+		return err;
+
+	*val = pl_float(&opt);
+
+	return 0;
+}
+
+
+static void enc_destructor(void *arg)
+{
+	struct augain_enc *st = arg;
+
+	list_unlink(&st->af.le);
+}
+
+
+static int encode_update(struct aufilt_enc_st **stp, void **ctx,
+			 const struct aufilt *af, struct aufilt_prm *prm,
+			 const struct audio *au)
+{
+	struct augain_enc *st;
+	(void)ctx;
+	(void)af;
+	(void)au;
+
+	if (!stp || !prm)
+		return EINVAL;
+
+	if (prm->fmt != AUFMT_S16LE) {
+		warning("augain: format not supported (%s)\n",
+			aufmt_name(prm->fmt));
+		return ENOTSUP;
+	}
+
+	st = mem_zalloc(sizeof(*st), enc_destructor);
+	if (!st)
+		return EINVAL;
+
+	conf_get_float(conf_cur(), "augain", &gain);
```

Comment:
can be done in init function

Reply:
Done.  It would have made sense in encode_update if there would be possibility to set the config var on the fly during call.  But I didn't find any conf_set functions.